src/tx/builder/index.ts

```diff
@@ -0,0 +1,579 @@
+    (
+      acc,
+      [key, fieldType, prefix],
+      index
+    ) => Object.assign(acc, { [key]: deserializeField(binary[index], fieldType, prefix) }),
+    {}
+  )
+}
+
+export interface BuiltTx<Tx extends TxSchema, Prefix extends EncodingType> {
+  tx: EncodedData<Prefix>
+  rlpEncoded: Uint8Array
+  binary: Uint8Array | NestedUint8Array
+  txObject: Tx
+}
+
+export type TxParamsBuild = TxParamsCommon & {
+  denomination?: AeAmountFormats
+}
+/**
+ * Build transaction hash
+ * @function
+ * @alias module:@aeternity/aepp-sdk/es/tx/builder
+ * @param params Object with tx params
+ * @param type Transaction type
+ * @param [options={}] options
+ * @param [options.excludeKeys] Array of keys to exclude for validation and build
+ * @param [options.prefix] Prefix of transaction
+ * @throws {Error} Validation error
+ * @returns object
+ * @returns object.tx Base64Check transaction hash with 'tx_' prefix
+ * @returns object.rlpEncoded rlp encoded transaction
+ * @returns object.binary binary transaction
+ */
+export function buildTx<Prefix> (
+  params: TxParamsBuild,
+  type: TxType,
+  { excludeKeys = [], prefix = 'tx', vsn, denomination = AE_AMOUNT_FORMATS.AETTOS }: {
+    excludeKeys?: string[]
+    prefix?: EncodingType
+    vsn?: number
+    denomination?: AeAmountFormats
+  } = {}
+): Prefix extends EncodingType
+    ? BuiltTx<TxSchema, Prefix>
+    : BuiltTx<TxSchema, 'tx'> {
+  const schemas = TX_SERIALIZATION_SCHEMA[type]
+  if (schemas == null) throw new UnexpectedTsError()
+
+  vsn ??= Math.max(
+    ...Object.keys(schemas).map(a => +a))
+  const schema = schemas[vsn as keyof typeof TX_SERIALIZATION_SCHEMA[typeof type]]
+  if (schema == null) {
+    throw new SchemaNotFoundError('serialization', type.toString(), vsn)
+  }
+
+  const tags = Object.entries(OBJECT_ID_TX_TYPE).find(([, t]) => t === type)
+  if (tags == null) { throw new UnexpectedTsError() }
+  const tag = tags[0]
+  const binary = buildRawTx(
+    { ...params, VSN: vsn, tag },
+    schema,
+    { excludeKeys, denomination: params.denomination ?? denomination }
+  ).filter(e => e !== undefined)
+
+  const rlpEncoded = rlpEncode(binary)
+  const tx = encode(rlpEncoded, prefix)
+  return {
+    tx,
+    rlpEncoded,
+    binary,
+    txObject: unpackRawTx(binary, schema)
+  } as any
+}
+
+export interface TxUnpacked<Tx extends TxSchema> {
+  txType: TxType
+  tx: RawTxObject<Tx>
+  rlpEncoded: Uint8Array
+  binary: Uint8Array | NestedUint8Array
+}
+/**
+ * Unpack transaction hash
+ * @function
+ * @alias module:@aeternity/aepp-sdk/es/tx/builder
+ * @param encodedTx String or RLP encoded transaction array
+ * (if fromRlpBinary flag is true)
+ * @param fromRlpBinary Unpack from RLP encoded transaction (default: false)
+ * @param prefix - Prefix of data
+ * @returns object
+ * @returns object.tx Object with transaction param's
+ * @returns object.rlpEncoded rlp encoded transaction
+ * @returns object.binary binary transaction
+ */
+export function unpackTx<TxType> (
```

Comment:
```suggestion
export function unpackTx<TxType extends keyof TxTypeSchemas> (
```
and to remove the switch at 528, not sure why it is needed